src/wsl_tools/__init__.py

```diff
@@ -63,6 +62,22 @@ def from_dotdesktop(cls, app_def: Path) -> Optional[WSLApp]:
         icon = de.getIcon()
         if name:
             return cls(name, generic_name, cmd, gui, icon)
+        else:
```

Comment:
Instead of duplicating the code here and bring the `WSLDistro` in this context, I prefer to move the check in the `WSLDistro.apps` property; here we can `raise IOError` instead of `return None`, then in `WSLDistro.apps` use `try...except IOError` to trigger the path conversion and call `WSLApp.from_dotdesktop` again.